xlsx/src/import/worksheets/mod.rs

```diff
@@ -0,0 +1,535 @@
+            let text = comment
+                .descendants()
+                .filter(|n| n.has_tag_name("t"))
+                .map(|n| n.text().unwrap().to_string())
+                .collect::<Vec<String>>()
+                .join("");
+            let cell_ref = get_attribute(&comment, "ref")?.to_string();
+            // TODO: Read author_name from the list of authors
+            let author_name = "".to_string();
+            comments.push(Comment {
+                text,
+                author_name,
+                author_id: None,
+                cell_ref,
+            });
+        }
+    }
+
+    Ok(comments)
+}
+
+fn get_formula_index(formula: &str, shared_formulas: &[String]) -> Option<i32> {
+    for (index, f) in shared_formulas.iter().enumerate() {
+        if f == formula {
+            return Some(index as i32);
+        }
+    }
+    None
+}
+
+fn from_a1_to_rc(
+    formula: String,
+    worksheets: &[String],
+    context: String,
+    tables: HashMap<String, Table>,
+) -> Result<String, XlsxError> {
+    let mut parser = Parser::new(worksheets.to_owned(), tables);
+    let cell_reference =
+        parse_reference(&context).map_err(|error| XlsxError::Xml(error.to_string()))?;
+    let t = parser.parse(&formula, &Some(cell_reference));
+    Ok(to_rc_format(&t))
+}
+
+#[derive(Error, Debug, PartialEq)]
+enum ParseReferenceError {
+    #[error("RowError: {0}")]
+    RowError(ParseIntError),
+    #[error("ColumnError: {0}")]
+    ColumnError(String),
+}
+
+// This parses Sheet1!AS23 into sheet, column and row
+// FIXME: This is buggy. Does not check that is a valid sheet name
+// There is a similar named function in ironcalc_base. We probably should fix both at the same time.
+// NB: Maybe use regexes for this?
+fn parse_reference(s: &str) -> Result<CellReferenceRC, ParseReferenceError> {
+    let bytes = s.as_bytes();
+    let mut sheet_name = "".to_string();
+    let mut column = "".to_string();
+    let mut row = "".to_string();
+    let mut state = "sheet"; // "sheet", "col", "row"
+    for &byte in bytes {
+        match state {
+            "sheet" => {
+                if byte == b'!' {
+                    state = "col"
+                } else {
+                    sheet_name.push(byte as char);
+                }
+            }
+            "col" => {
+                if byte.is_ascii_alphabetic() {
+                    column.push(byte as char);
+                } else {
+                    state = "row";
+                    row.push(byte as char);
+                }
+            }
+            _ => {
+                row.push(byte as char);
+            }
+        }
+    }
+    Ok(CellReferenceRC {
+        sheet: sheet_name,
+        row: row.parse::<i32>().map_err(ParseReferenceError::RowError)?,
+        column: column_to_number(&column).map_err(ParseReferenceError::ColumnError)?,
+    })
+}
+
+fn get_column_from_ref(s: &str) -> String {
+    s.chars().filter(|c| !c.is_ascii_digit()).collect()
+}
```

Comment:
This surprisingly was a big perf win compared to the old version:
```
    let cs = s.chars();
    let mut column = Vec::<char>::new();
    for c in cs {
        if !c.is_ascii_digit() {
            column.push(c);
        }
    }
    column.into_iter().collect()
    ```
    
    That old version had a lot more allocations to grow the vector, and this method is called very often.